node/network/statement-distribution/src/lib.rs

```diff
@@ -208,9 +211,12 @@ impl<R: rand::Rng> StatementDistributionSubsystem<R> {
 				}
 
 				if let Some(activated) = activated {
-					// TODO [now]: legacy, activate only if no prospective parachains support.
-					crate::legacy_v1::handle_activated_leaf(ctx, legacy_v1_state, activated)
-						.await?;
+					// Legacy, activate only if no prospective parachains support.
+					let mode = prospective_parachains_mode(ctx.sender(), activated.hash).await?;
+					if let ProspectiveParachainsMode::Disabled = mode {
```

Comment:
Do we have to do the same for `deactivated` above, or does it not matter, since `deactivate_leaves` only affects the local state?

Reply:
https://github.com/paritytech/polkadot/blob/9f27a2f0e82ab1860e36e2ec45c88afa0209d1f2/node/network/statement-distribution/src/legacy_v1/mod.rs#L1988

If a leaf wasn't activated prior, this does nothing, so can be ignored.